### ISSUE TITLE
Support BIGTIFF

### ISSUE THREAD

**Issue by dmahr1** [geospatial-jeff/aiocogeo#103]
Pretty self exaplantory, aiocogeo doesn't currently support BIGTIFF. https://github.com/geospatial-jeff/aiocogeo/blob/950ea55e3a885bbd3b9123ebc51c398403e13d44/aiocogeo/cog.py#L88

**Comment by dmahr1** [comment]
@geospatial-jeff I would be interested in working on this. We're hoping to productionize COGs at DroneDeploy in a few months and BIGTIFF support would be very nice to have. I'm not an expert with the TIFF specification though...is this just a matter of doubling the number of bytes read for each offset or is there more complexity lurking in the shadows? I suppose I can look at the GDAL/libtiff implementation?

**Comment by geospatial-jeff** [comment]
> is this just a matter of doubling the number of bytes read for each offset

AFAIK this is correct, but I am also not an expert in BIGTIFF

Here are some other implementations to look at in addition to GDAL/libtiff.  I would start with COGDumper since the initial aiocogeo code was based on that project.
- COGDumper - https://github.com/mapbox/COGDumper
- geotiff.js - https://github.com/geotiffjs/geotiff.js

**Comment by mykolakozyr** [comment]
@dmahr1 have you had a chance to work on this topic? Use-cases look exciting, would be happy to join.

**Comment by dmahr1** [comment]
:wave: @mykolakozyr I have not had a chance to start on this. There was a change in priorities at my work so I've been busy with other things. It would be great if you want to give it a try!

**Comment by kylebarron** [comment]
I think @geospatial-jeff might have mentioned there could be breaking changes soon? Might be best to wait to hear back from him before starting work on this.

**Comment by rdenham** [comment]
Not sure if anybody still needs this or not, but I needed something like this for some images I was accessing. I had trouble understanding how it all worked, so was reluctant to make pull request. Instead, to allow me to understand it, I wrote my own package, available at https://gitlab.com/jrsrp/sys/asyncog. It's pretty much a crappier and more limited version of  this package, but I learned a lot about COGs on the way. But maybe it would be useful to others, even if just to encourage someone to do this the right way, and add BigTiff support to aiocogeo.
